homeassistant/components/nuki/binary_sensor.py

```diff
@@ -19,16 +19,22 @@
 async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
-    """Set up the Nuki lock binary sensor."""
+    """Set up the Nuki binary sensors."""
     entry_data: NukiEntryData = hass.data[NUKI_DOMAIN][entry.entry_id]
 
-    entities = []
+    lock_entities = []
+    opener_entities = []
 
     for lock in entry_data.locks:
         if lock.is_door_sensor_activated:
-            entities.extend([NukiDoorsensorEntity(entry_data.coordinator, lock)])
+            lock_entities.extend([NukiDoorsensorEntity(entry_data.coordinator, lock)])
 
-    async_add_entities(entities)
+    async_add_entities(lock_entities)
+
+    for opener in entry_data.openers:
+        opener_entities.extend([NukiRingactionEntity(entry_data.coordinator, opener)])
+
+    async_add_entities(opener_entities)
```

Comment:
Please make a single call to add the entities.

Reply:
https://github.com/home-assistant/core/pull/109696